app/ide-desktop/lib/dashboard/src/authentication/src/authentication/providers/session.tsx

```diff
@@ -5,6 +5,23 @@ import * as results from "ts-results";
 
 import * as hooks from "../../hooks";
 import * as cognito from "../cognito";
+import * as listen from "../listen";
+
+// =================
+// === Constants ===
+// =================
+
+/** URL that the Electron main page is hosted on.
+ *
+ * This **must** be the actual page that the Electron app is hosted on, otherwise the OAuth flow
+ * will not work and will redirect the user to a blank page. If this is the correct URL, no redirect
+ * will occur (which is the desired behaviour). */
+const MAIN_PAGE_URL = "http://localhost:8080";
```

Comment:
1. Electron app is a system app and it is not hosted on a page.
2. Why are we hardcoding port here? The content of the Electron app is served on different ports, depending on their availability. I was writing so many comments about it in previous PRs. Please do not hardcode ports in these PRs.

Reply:
The Electron app is served from a web view. The page must remain on that web view to avoid reloading the whole application. Amplify, annoyingly, attempts to redirect the user off of the web view. It seems the only way to prevent that is to force to stay on the same page by writing the original URL back to the window state.

Our options are to either:
1. Disable Amplify's behaviour -- but I haven't found a way to do so.
2. Pass the port that we chose to open the web view on to the Dashboard -- we can pass props from `lib/content` to `lib/dashboard`, but it's not `lib/content` that knows the port we're running on. It's one of `lib/client` or `lib/server`. How do we pass the port to `lib/content`?
3. Detect the port we're running on by reading it from the `window.location` when the Dashboard starts, prior to authentication.

I think #3 is probably the best bet for loose coupling.

Reply:
Both solutions 2 and 3 seems feasible. You can pass the info from `client` to `content` via RPC protocol (you guys have been using RPC electron protocol already in these PRs). Although, I think the solution 3 is cleaner and simpler indeed.

Reply:
Please ping me when that will be fixed :)